sdk/ai/azure-ai-projects/README.md

```diff
@@ -0,0 +1,580 @@
+```
+
+#### Prequisites
+
+- Python 3.8 or later is required to use this package.
+- You need an [Azure subscription][azure_sub] to use this package.
+- An existing Azure Ai Projects instance.
+#### Create with an Azure Active Directory Credential
+To use an [Azure Active Directory (AAD) token credential][authenticate_with_token],
+provide an instance of the desired credential type obtained from the
+[azure-identity][azure_identity_credentials] library.
+
+To authenticate with AAD, you must first [pip][pip] install [`azure-identity`][azure_identity_pip]
+
+After setup, you can choose which type of [credential][azure_identity_credentials] from azure.identity to use.
+As an example, [DefaultAzureCredential][default_azure_credential] can be used to authenticate the client:
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+`AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`
+
+Use the returned token credential to authenticate the client:
+
+```python
+>>> from azure.ai.projects import AIProjectClient
+>>> from azure.identity import DefaultAzureCredential
+>>> client = AIProjectClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
+```
+
+## Key concepts
+
+TODO
+
+## Examples
+
+### Agents
+The following steps outline the typical sequence for interacting with agents:
+
+  - <a href='#create-project-client'>Create a project client</a>
+  - <a href='#create-agent'>Create an agent</a> with <a href='#create-agent-with-toolset-or-tools-and-tool-resources'>toolset, or tools and tool resources</a> including:
+    - <a href='#create-agent-with-file-upload-in-vector-store-for-file-search'>File Search with file upload indexed by vector stores</a>
+    - <a href='#create-agent-with-file-upload-for-code-interpreter'>Code Interpreter with file upload</a>
+    - <a href='#create-agent-with-function-tool'>Function calls</a>
+  - <a href='#create-thread'>Create a thread</a> with
+     - <a href='#create-thread-with-tool-resource'>Tool resource</a>
+  - <a href='#create-message'>Create a message</a> with:
+    - <a href='#create-message-with-file-search-attachment'>File search attachment</a>
+    - <a href='#create-message-with-code-interpreter-file-attachment'>Code interpreter attachment</a>
+  - <a href='#create-run-run_and_process-or-stream'>Execute Run, Run_and_Process, or Stream</a>
+  - <a href='#retrieve-messages'>Retrieve messages</a>
+  - <a href='#teardown'>Tear down by deleting resources</a>
+
+
+#### Create Project Client
+
+When you create a project client, you need to make the decision to use synchronous or asynchronous client. Use either:
+
+```python
+from azure.ai.projects import AIProjectClient
+# OR
+from azure.ai.projects.aio import AIProjectClient
+```
+
+Here is an example of creating a project client:
+<!-- SNIPPET:sample_agents_basics.create_project_client -->
+
+```python
+project_client = AIProjectClient.from_connection_string(
```

Comment:
this doesnt look proper.... why would I need a credential with connection string ?
in general for samples we should be using credential rather than connection_string

Reply:
@kashifkhan thank you for your comments! Connection string uniquely defines your AI Studio Project. It does not include credentials. Therefore, we need both. Feel free to start an IM thread with me (Darren Cohen) and Johan to discuss this. But this is how we are going to ship the first beta for Ignite.